[bug 1319244] test Activist cross-site search

craigcook opened the pull request:
## Description
Adds a Google custom search box to a few pages:
* /about
* /about/manifesto
* /foundation/*

This is the "Activist" search indexing sites mostly focused on advocacy, web education, internet health, etc.

The test will be English only and uses the switch `mozorg-search-activist`.

## Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1319244

## Testing
Mostly visual, since we're doing some CSS hackery to customize the look of their search box. Also verify searching works as expected and doesn't throw JavaScript errors.

craigcook commented:
Marking Do Not Merge, going to set up traffic cop and a conditional so we can control how many users see it.

bensternthal commented:
Testing js error on this page (could be others) and search does not render.

https://bedrock-demo-search.us-west.moz.works/en-US/foundation/issues/

craigcook commented:
> Testing js error on this page (could be others) and search does not render.

Fixed. I changed an ID in one place but hadn't updated the other pages.

jpetto commented on bedrock/foundation/templates/foundation/base-resp.html:
```diff
@@ -30,6 +30,7 @@
 {% block content %}
   <main>
     <header class="page-header">
+      {% block search %}search{% endblock %}
```
Should leave the block empty here so "search" doesn't show up on other pages (e.g. `/foundation/advocacy/`).

craigcook replied:
D'oh, artifact from testing. My bad!

jpetto commented on bedrock/settings/base.py:
```diff
@@ -584,7 +584,7 @@ def lazy_email_backend():
 
 # Google Analytics
-GA_ACCOUNT_CODE = ''
+GA_ACCOUNT_CODE = 'UA-36116321-20'
```
Do we want to put this account code here? Should probably go in `.env` locally, right?

bensternthal replied:
I think you copied this from my test, I left this in by accident when I was testing. I think this should be completely removed.

Double check this should be nix'd with paul tomorrow when things are less crazy.

craigcook replied:
Yep, it was just copied over, wasn't sure if it was needed or not. I'll remove.

jpetto commented on media/js/base/search-activist.js:
```diff
@@ -0,0 +1,57 @@
+            if (e.keyCode === 13 || e.which === 13) { // Enter
+                setGTMSearch();
+            }
+        });
```
Not sure if it's supported/possible, but would be nice to have `Escape` close the modal.

craigcook replied:
I started working on that, as well as setting focus to the modal when it opens, but it proved difficult and was eating my time. I think we should add it as a follow-up enhancement if we can.

jpetto commented on media/js/base/search-activist.js:
```diff
@@ -0,0 +1,57 @@
+    changeTerm();
+
+    displayTerm.on('click', function() {
+        $('input.gsc-input').val(displayTerm.text());
```
Nit - should cache `$('input.gsc-input')` so jQuery doesn't have to look it up each time.

jpetto commented on media/css/pebbles/components/google-search.scss:
```diff
@@ -0,0 +1,113 @@
+    .gsc-result .gs-title {
+        height: auto !important;
+    }
+}
```
The modal begins to get off center around 700px wide. Maybe add a media query?

![screen shot 2016-11-30 at 4 28 31 pm](https://cloud.githubusercontent.com/assets/317532/20774014/10a58236-b71a-11e6-9fbb-f879ef0fc89c.png)

jpetto commented on bedrock/mozorg/templates/mozorg/about/manifesto.html:
```diff
@@ -52,9 +58,21 @@
+        <label class="search-label" for="gsc-i-id1"><span class="search-label-prefix">{{ _('Search Mozilla:') }}</span> <span class="search-label-term">{{ _('Cybersecurity') }}</span></label>
+        <gcse:search gname="mozsearch"></gcse:search>
+      </div>
+    </aside>
```
Maybe make this `<aside>` an include for DRY-ness?

craigcook replied:
That's the plan, but I was waiting until I start building out the other two searches. I want to make it a macro that accepts a few arguments for a modifier class, suggested terms, etc. I suppose for now it could just be a static include though.

jpetto commented:
LGTM, even in IE 9. r+ 🕵️
